cmd/bucket-replication.go

```diff
@@ -2284,7 +2284,7 @@ func proxyHeadToRepTarget(ctx context.Context, bucket, object string, rs *HTTPRa
 			DeleteMarker:              objInfo.IsDeleteMarker,
 			ContentType:               objInfo.ContentType,
 			Expires:                   objInfo.Expires,
-			StorageClass:              objInfo.StorageClass,
+			StorageClass:              globalAPIConfig.storageClass(objInfo.StorageClass),
```

Comment:
We should return what the replica site returns here. In the scenario where the user has applications sensitive to the value returned in `X-Amz-Storage-Class` header, `api.force_storage_class` must be configured on the replica too.